File: src/org/webfabric/io/RelativeResourceLoader.java
package org.webfabric.io;

import java.io.InputStream;

public class RelativeResourceLoader {
    private final Class aClass;

    public RelativeResourceLoader(Class aClass) {
        this.aClass = aClass;
    }

    public InputStream load(String relativeFilename) {
        String name = aClass.getPackage().getName();
        name = name.replace(".", "/");
        String resource = name + "/" + relativeFilename;
        return aClass.getClassLoader().getResourceAsStream(resource);
    }
}

File: src/org/webfabric/io/RelativeResourceLoader.scala
package org.webfabric.io

;

import scala.reflect.Manifest
import java.io.InputStream;

object RelativeResourceLoader {
  def load[T](relativeFilename: String)(implicit manifest: Manifest[T]): InputStream = {
    val aClass = manifest.erasure
    var name = aClass.getPackage().getName();
    name = name.replace(".", "/");
    val resource = name + "/" + relativeFilename;
    return aClass.getClassLoader().getResourceAsStream(resource);
  }
}
